cli/tests/activate.bats

```diff
@@ -2081,3 +2081,84 @@ EOF
 }
 
 # ---------------------------------------------------------------------------- #
+
+# bats test_tags=activate,activate:nested_flox_activate_tracelevel
+@test "{bash,fish,tcsh,zsh}: confirm _flox_activate_tracelevel set in nested activation" {
+  project_setup
+
+  # The shell-specific flox init scripts finish by unsetting the
+  # _flox_activate_tracelevel environment variable, and this can
+  # cause problems for an "outer" interactive activation when there
+  # is an "inner" in-place activation happening by way of a "dotfile".
+
+  # Set up this test by creating dotfiles which perform an in-place
+  # activation, and then run an interactive activation of a second
+  # environment to confirm that _flox_activate_tracelevel is set
+  # for the outer activation.
+
+  # Start by adding logic to create semaphore files for all shells.
+  for i in "$HOME/.bashrc.extra" "$HOME/.config/fish/config.fish.extra" "$HOME/.tcshrc.extra" "$HOME/.zshrc.extra"; do
+    cat > "$i" <<EOF
+touch "$PROJECT_DIR/_flox_activate_tracelevel.in_test"
+test -n "\$_flox_activate_tracelevel" || touch "$PROJECT_DIR/_flox_activate_tracelevel.not_defined"
+EOF
+  done
+
+  # Finish by appending shell-specific flox activation syntax.
+  for i in "$HOME/.bashrc.extra" "$HOME/.config/fish/config.fish.extra" "$HOME/.zshrc.extra"; do
+    echo "eval \"\$($FLOX_BIN activate --dir $PROJECT_DIR)\"" >> "$i"
+  done
+  echo "eval \"\`$FLOX_BIN activate --dir $PROJECT_DIR\`\"" >> "$HOME/.tcshrc.extra"
+
+  # Create a test environment.
+  _temp_env="$(mktemp -d)"
+  "$FLOX_BIN" init -d "$_temp_env"
+
+  # Activate the test environment from each shell, each of which will
+  # launch an interactive shell that sources the relevant dotfile.
+  echo "Testing bash"
```

Comment:
Non-blocking, commentary: I really wish bats made it easier to do table-style tests for differentiating the small differences between these test cases.

Reply:
👍 Agreed. I too find this suboptimal, but workable, and haven't come up with anything better.